src/constants/objectHelpers.ts

```diff
@@ -102,24 +103,29 @@ const parseRecordToObjects = (or: ObjectRecord): ObjectsForTables => {
 
 /** Takes the list of constituent records and transforms them by trimming out any unneeded fields that were included in the object */
 const createListOfConstituentRecordObjects = (
-  constituentRecords: ObjectRecord["ConstituentRecord"]
+  constituentRecords: ObjectRecord["ConstituentRecord"],
+  objectId: number
 ): {}[] => {
-  return constituentRecords.map((cr) => {
+  return constituentRecords.map((constituentRecord) => {
     // Create the empty constituent record object
     const constituentRecordObject = {};
 
     // Check if the current field name is needed in the constituent record object
-    for (let [key, value] of Object.entries(cr)) {
-      const fieldNameNeededInCR = NetXTables.constituentRecords.columns.some(
-        (column) => column.name === key
-      );
-
-      // If it is, add it
-      if (fieldNameNeededInCR) {
-        constituentRecordObject[key] = value;
+    for (let [fieldName, fieldValue] of Object.entries(constituentRecord)) {
+      const fieldNameIsNeededInConstituentRecord =
+        NetXTables.constituentRecords.columns.some(
+          (column) => column.name === fieldName
+        );
+
+      // If it is a field we need, add it
+      if (fieldNameIsNeededInConstituentRecord) {
+        constituentRecordObject[fieldName] = fieldValue;
       }
     }
 
+    // We are also adding the Object Id that this constituent record relates to
+    constituentRecordObject["objectId"] = objectId;
```

Comment:
Each constituent record actually is associated with an object - making each constituent record unique to an object rather than globally reusable by each object that was created by the constituent